planet/clients/orders.py

```diff
@@ -337,71 +340,90 @@ async def download_order(
-        If many orders are being polled asynchronously, consider
-        increasing the wait to avoid throttling.
+        """Wait until order reaches one of the specified states.
+
+        This function polls the Orders API to determine the order state, with
+        the specified delay between each polling attempt, until the
+        order reaches one of the set of desired states. If the maximum number
+        of attempts is reached before the order reaches one of the desired
+        states, an exception is raised. Setting 'max_attempts' to zero will
+        result in no limit on the number of attempts.
+
+        Setting 'delay' to zero results in no delay between polling attempts.
+        This will likely result in throttling by the Orders API, which has
+        a rate limit of 10 requests per second. If many orders are being
+        polled asynchronously, consider increasing the delay to avoid
+        throttling.
+
+        By default, the set of states that result are polled for is the
+        set of completed states. This can be changed to any set of valid
+        order states. It is not recommended to change this to a subset of
+        completed states, as this opens up the possibility of the the
+        specified state never being reached (i.e. if the order completes in
+        a 'partial' state but only the 'success' state was specified).
 
         Example:
             ```python
             from planet.reporting import StateBar
 
             with StateBar() as bar:
-                await poll(order_id, report=bar.update)
+                await wait(order_id, report=bar.update_state)
             ```
 
         Parameters:
             order_id: The ID of the order
-            state: State to poll until. If multiple, use list. Defaults to
-                any completed state.
-            wait: Time (in seconds) between polls.
-            report: Callback function for progress updates. Invoked with
-                keyword arguments `state` (poll state) and `logger`
-                (callback for logging progress bar status). Recommended
-                value is `reporting.StateBar.update`.
+            states: Order states that will end polling. Defaults to all
+                completed states.
+            delay: Time (in seconds) between polls.
+            max_attempts: Maximum number of polls. Set to zero for no limit.
+            report: Callback function for reporting progress updates.
 
         Returns
-            Completed state of the order.
+            State of the order.
 
         Raises:
             planet.exceptions.APIException: On API error.
-            OrdersClientException: If order_id is not valid or state is not
-                supported.
+            planet.exceptions.ValueError: If order_id or one or more of the
+                states are not valid.
+            planet.exceptions.MaxAttemptsError: If the maximum number of
```

Comment:
Yep, we need an error like this. MaxAttemptsReachedError might be more descriptive? Or SickOfWaitingError :laughing:

Reply:
LOL or the more general WTFError. I can change it to MaxAttemptsReachedError.

Reply:
actually, what about [RetriesExceededError](https://github.com/boto/boto3/blob/c236fa1d230ea2f4738d8d5e5961d587ef006299/boto3/exceptions.py#L70)?

Reply:
just falling back to ClientError for now